### DISCUSSION TITLE
Setting a toolTip

### DISCUSSION THREAD

**Discussion by AviadBly** [AvaloniaUI/Avalonia#12018] [Q&A] [answered]
Hi, I'm new to mvvm and avalonia ui. Wanted to know how to set a tooltip in the c# codebehind?
This is an example in WinForms and I would like to know how to reformat it in Avalonia. 

VideoPix2PtResultToolTip.SetToolTip(VideoImagePictureBox, $"{pix2PtAnswerStr}");

**Comment by timunie** [comment, marked as answer]
In MVVM, you just can do:

```xml
<Button Content="ToolTip" ToolTip.Tip="{Binding MyTooltip}" />
```

**Reply by AviadBly** [reply]
Hi, Thanks for replying.
But what if I want that action to trigger only if conditional is met?

I want something like this to happen:

else if (e.GetCurrentPoint(this).Properties.IsRightButtonPressed)
            {
                VideoPix2PtResultToolTip.SetToolTip(VideoImagePictureBox, $"x:{clickPosition.X},y:{clickPosition.Y}");
            }

**Comment by jmnicolas90** [comment]
Hi, you can add a Tooltip in code behind this way:

`ToolTip.SetTip(myButton, "This is my button tooltip");`

**Reply by NickBarton** [reply]
And you can unset it in this way.

myButton.ClearValue(ToolTip.TipProperty);

**Reply by NinjaPedroX** [reply]
> And you can unset it in this way.
> 
> myButton.ClearValue(ToolTip.TipProperty);

Following that format, you can set the tooltip like this as well:

`myButton.SetValue(ToolTip.TipProperty, "Tooltip text");`
